src/main/java/helper/DateTimeHelper.java

```diff
@@ -0,0 +1,126 @@
+package helper;
+
+import exception.InvalidDateFormatException;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+
+/**
+ * Represents the helper class to abstract LocalDateTime logic in Duke.
+ */
+public class DateTimeHelper {
+    /**
+     * Returns corresponding LocalDateTime object for the given string. This
+     * method is used when receiving user input.
+     * The string must be in the format DD/MM/YYYY HHMM where the year can
+     * be any number of digits > 0 and each time-based unit (eg. day, month)
+     * must be valid.
+     *
+     * @param dateTimeString A string representing the date.
+     * @return The corresponding datetime object for the string.
+     * @throws InvalidDateFormatException If the string does not follow the specified format.
+     */
+    public static LocalDateTime parse(String dateTimeString) throws InvalidDateFormatException {
+        // Check if string is in the correct format.
+        if (!dateTimeString.matches("\\d{1,3}/\\d{1,3}/\\d+ \\d{4}")) {
+            throw new InvalidDateFormatException("Incorrect date format!");
+        }
+
+        // Parse day, month, year, hour and minute from dateTimeString.
+        String[] values = dateTimeString.split("/");
+        Integer day = Integer.valueOf(values[0]);
+        Integer month = Integer.valueOf(values[1]);
+
+        String[] yearAndTime = values[2].split(" ");
+        Integer year = Integer.valueOf(yearAndTime[0]);
+        Integer hour = Integer.valueOf(yearAndTime[1].substring(0, 2));
+        Integer minute = Integer.valueOf(yearAndTime[1].substring(2));
+
+        // Convert the time units into a LocalDateTime object.
+        try {
+            return LocalDateTime.of(year, month, day, hour, minute);
```

Comment:
Unrelated to coding standards, but you can check out `LocalDateTime.parse`

You can pass it the specific format you want and it'll also throw a `DateTimeParseException` if the string is in the wrong format, so you can save the trouble of validating it yourself.

[Here's the docs](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#parse-java.lang.CharSequence-java.time.format.DateTimeFormatter-) if you want to read about it.